### ISSUE TITLE
switch sink by name

### ISSUE THREAD

**Issue by root-hal9000** [mpapis/home_dotfiles#4]
Thanks for the audio switch script. I am a little confused about the switching by name part, as I am trying to use that one to add some keyboard shortcuts (the number ids change when I reconnect my bluetooth speakers). when doing it from the command line by name, I have tried several different formats without success - what I am doing wrong here? I guess I don;t know how to pass the name string? Here are some of my attempts

```
$ ./audio.sh alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3
Usage: ./audio.sh [gui|list|<sink name to switch to>]
$ ./audio.sh <"alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3">
bash: syntax error near unexpected token `newline'
$ ./audio.sh <'alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3'>bash: syntax error near unexpected token `newline'
$ ./audio.sh <alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3>
bash: syntax error near unexpected token `newline'
$ ./audio.sh 'alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3'
Usage: ./audio.sh [gui|list|<sink name to switch to>]
$ ./audio.sh "alsa_output.pci-0000_42_00.1.hdmi-stereo-extra3"
Usage: ./audio.sh [gui|list|<sink name to switch to>]
```

**Comment by root-hal9000** [comment]
also tried removing alsa_output as well as the other prefixes in the name but no dice...

**Comment by mpapis** [comment]
Now, where is the script? Can't find it ;)

**Comment by root-hal9000** [comment]
wait, what? Your [audio switch script](https://github.com/mpapis/home_dotfiles/blob/4a1b30e36b6e6772d4e99050aca740c14b460aeb/bin/mypa)? From the winky emoji I feel like I am doing/saying something really stupid but I don;t get it
PS.: in my home folder, where I am running it successfully but not able to switch by string name, only by output ID, which unfortunately does not persist for my bluetooth devices

**Comment by mpapis** [comment]
I can not find audio.sh

**Comment by root-hal9000** [comment]
I am sorry @mpapis , I just realized I renamed it locally but it is the same file named mypa that I linked above. It is running as intended as you can see from the output above, I just can't do it with the string name parameters that it suggests in its usage line that prints out when run without parameters... Thanks!

**Comment by root-hal9000** [comment]
In any case, I will try to mess with the regular command to see 8f I can work out the string issue and edit your script accordingly if I can get it to work that way, in which case I'll share what I did

**Comment by mpapis** [comment]
aha! try `mypa "<alsa_output.pci-0000_00_1f.3.analog-stereo>"`

**Comment by root-hal9000** [comment]
no, in that case I still get just the Usage help text. I also have some other examples of what I tried in the original post.

In my ignorance, looking at the case statement at the bottom of the file, I don't see an option for handling the text parameter input - or am I wrong?
```
case "${1:-}" in
  (""|list) list_sinks       ;;
  ([0-9]*)  switch_sink "$@" ;;
  (gui)     switch_gui       ;;
  (*)       help_me          ;;
esac
```

sorry to bother with this, I know it's an old thing for you
